File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string) : string =
    let parts = logLine.Split ":"
    parts[1].Trim()

let logLevel (logLine: string) : string =
    let parts = logLine.Split ":"
    parts[0].Replace("[", "").Replace("]", "").Trim().ToLower()

let reformat (logLine: string) : string =
    sprintf "%s (%s)" (message logLine) (logLevel logLine)

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
  public static string Message(string logLine) =>
    logLine.Split(":", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)[1];

  public static string LogLevel(string logLine) =>
    logLine.Split(":", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)[0][1..^1].ToLower();

  public static string Reformat(string logLine) =>
    $"{Message(logLine)} ({LogLevel(logLine)})";
}
